app/server/lib/ExternalStorage.ts

```diff
@@ -58,7 +58,11 @@ export interface ExternalStorage {
   // Close the storage object.
   close(): Promise<void>;
 
-  uploadStream?(key: string, inStream: stream.Readable, metadata?: ObjMetadata): Promise<string|null|typeof Unchanged>;
+  uploadStream?(key: string,
+                inStream: stream.Readable,
+                size?: number,
+                metadata?: ObjMetadata
+  ): Promise<string|null|typeof Unchanged>;
```

Comment:
This line should be adapted after this change:
https://github.com/gristlabs/grist-core/blob/ec0b88572333b87102e25bfa6a78e00b101a578c/app/server/lib/ExternalStorage.ts#L78-L79

Reply:
Also we might be interested in adapting the `ExternalStorageAttachmentStore` as well: 
https://github.com/gristlabs/grist-core/blob/ec0b88572333b87102e25bfa6a78e00b101a578c/app/server/lib/AttachmentStore.ts#L127

Reply:
I thought about it but decided againt it because :
- there's no way to know the size of the steam of data coming in (https://github.com/gristlabs/grist-core/pull/1489#issuecomment-2690981078)
- i'd want to keep this PR as small and manageable as possible